solution.cpp

```diff
@@ -22,113 +22,159 @@ using std::string;
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
+
+
+int write_composites( const int size_of,
                       bool primes[],
                       const string& infilename,
                       const string& outfilename );
 
+
+
 // ----------------------------------------------------------------------------
 
 int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
+
+    infilename.open("input.txt");
+
+    int MAX = find_max(infilename);
+    if(MAX == -1)
+    {
+        cout << "Error in find_max()." << endl;
+        return 1;
     }
 
-    bool primes[max+1];
-    sieve(max+1, primes);
+    bool primes[max_num + 1];
+    sieve(max_num + 1, primes);
 
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
+    int ret_value;
+    ret_value = write_primes(total,primes,infilename,outfilename);
+    if(ret_value == -1)
+    {
+        cout << "Error in write_primes()." << endl;
+        return 1;
     }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
+
+    int ret;
+    ret = write_composites(total,primes,infilename,outfilename);
+     if(ret_value == -1)
+    {
+        cout << "Error in write_composites()." << endl;
+        return 1;
     }
 
-    return 0;  // success
+    cout << "Files successfully created and sorted." << endl;
+
+    return 0;
 }
 
 // ----------------------------------------------------------------------------
 
-// TODO: write prototyped functions
+int find_max(ifstream& infilename)
+{
+    infilename.clear();
+    infilename.seekg(0, infilename.beg);
+    if(infilename){
+    int number;
+    total = 0;
+
+    while(infilename >> number)                      // Read through file from beginning. If the number is larger than
+    {                                                // the max_number set at 1, replace it to be the max_number.
+        total++;                                     // The min_number is to eliminate any numbers larger than the limit
+        if(number > max_num)                         // set at 65535.
+            max_num = number;
+        if(number < min_num)
+            min_num = number;
+    }
+    return max_num;
+    }
+    else
+        return -1;
+}
 
+void sieve(const int size_of, bool primes[])
+{
+    for(int TRUE = 0; TRUE < max_num + 1; TRUE++)     // Set all elements to true
```

Comment:
all caps names look like macros or constants.  also, naming a variable any variation on `true` or `false` is probably not the best idea.